File: cpp/1564.put-boxes-into-the-warehouse-i.cpp
class Solution {
public:
    int maxBoxesInWarehouse(vector<int>& boxes, vector<int>& warehouse) {
        sort(boxes.begin(), boxes.end(), greater<int>());
	int res = 0, minv=INT_MAX, j=0;
	int n = boxes.size();
	for(auto w:warehouse){
		minv = min(minv, w);
		while(j < n && boxes[j] > minv)j++;
		if(j<n){
			res++;
			j++;
		}
	}
	return res;
    }
};

File: python/1564.put-boxes-into-the-warehouse-i.py
class Solution:
    def maxBoxesInWarehouse(self, boxes: List[int], warehouse: List[int]) -> int:
         boxes.sort()
         res = 0
         j = len(boxes)-1
         for space in warehouse:
             while j >=0 and boxes[j] > space:
                 j-=1
             if j >=0:
                 res+=1
                 j-=1
         return res
